lib/src/entrypoint.dart

```diff
@@ -226,6 +226,38 @@ class Entrypoint {
     bool dryRun = false,
     bool precompile = false,
   }) async {
+    // We require an SDK constraint lower-bound as of Dart 2.12.0
+    final dartSdkConstraint = root.pubspec.sdkConstraints['dart'];
+    if (dartSdkConstraint is! VersionRange ||
+        (dartSdkConstraint is VersionRange && dartSdkConstraint.min == null)) {
+      // Suggest version range '>=2.10.0 <3.0.0', we avoid using:
+      // [CompatibleWithVersionRange] because some pub versions don't support
+      // caret syntax (e.g. '^2.10.0')
+      var suggestedConstraint = VersionRange(
+        min: Version.parse('2.10.0'),
+        max: Version.parse('2.10.0').nextBreaking,
```

Comment:
```suggestion
        min: Version.parse(Platform.version.split(' ').first),
        max: Version.parse(Platform.version.split(' ').first).nextBreaking,
```

Reply:
Then I could just use `sdk.version`.

For first few versions will recommend `2.10.0` as the lower-bound, we change this recommendation later...